base/src/Data/Macaw/AbsDomain/JumpBounds.hs

```diff
@@ -661,89 +661,154 @@ addRangePred cns v p =
         _ ->
           Right $ branchAssignRangePred n p
 
+-- | @addLowerBoundPred cns w l x@ adds a @l <= x@ constraint over the `w`-bit-truncated values.
+-- Fails when the bound excludes all possible values (lower bound greather than the maximum `w`-bit
+-- value).
+-- Note: The arguments are ordered so that it reads more naturally, as the difference of type
+-- between `l` and `x` prevents mistakes.
+addLowerBoundPred ::
+  RegisterInfo (ArchReg arch) =>
+  BlockIntraStackConstraints arch ids -> NatRepr n -> Integer -> Value arch ids (BVType n) ->
+  Either String (BranchConstraints arch ids)
+addLowerBoundPred cns w l x
+  | l >= maxUnsigned w = Left "Lower bound excludes all possible values"
```

Comment:
This code confuses me a bit, since the Haddocks claim that it should fail when the lower bound (`l`) is _greater_ than the maximum `w`-bit value (`maxUnsigned w`), but this code (`l >= maxUnsigned w`) will return `True` if `l` is equal to the maximum `w`-bit value. What am I missing?

Reply:
Correct, it should be `>`. When `l` is exactly `maxUnsigned w`, this is still a valid range of one possible value. Fixing.